### PR TITLE
docs: update for 5.0

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: docs/BUILD.bazel [hunk 1]
```diff
@@ -30,7 +37,7 @@ stardoc(
 stardoc(
     name = "providers",
     out = "providers.md_",
-    input = "//:providers.bzl",
+    input = "//nodejs:providers.bzl",
```
Comment (gregmagolan) [inline docs/BUILD.bazel:40]: Should still have the old @build_bazel_rules_nodejs providers in the docs and mke the new one CoreProviders?

Comment (alexeagle) [reply docs/BUILD.bazel:40]: I thought we were getting rid of the old providers in your change?

Comment (gregmagolan) [reply docs/BUILD.bazel:40]: They are still the ones exported from `@build_bazel_rules_nodejs//:providers.bzl` and still in use by the rules there
```
ExternalNpmPackageInfo = _ExternalNpmPackageInfo
js_ecma_script_module_info = _js_ecma_script_module_info
js_named_module_info = _js_named_module_info
JSEcmaScriptModuleInfo = _JSEcmaScriptModuleInfo
JSNamedModuleInfo = _JSNamedModuleInfo
NODE_CONTEXT_ATTRS = _NODE_CONTEXT_ATTRS
node_modules_aspect = _node_modules_aspect
NodeContextInfo = _NodeContextInfo
NodeRuntimeDepsInfo = _NodeRuntimeDepsInfo
run_node = _run_node
```

Comment (alexeagle) [reply docs/BUILD.bazel]: Fixed, the new providers are in Core.md and the old Providers.md still has the old ones, and also a link to Core

**Review by gregmagolan — APPROVED** [review]
🌮

**Pull request merged**